internal/resource/resource_test.go

```diff
@@ -213,3 +208,174 @@ func TestNewResource(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeBasics(t *testing.T) {
```

Comment:
The reconciliation controller has tests to cover most of this, but we should also expand test coverage here over time.